emily/handler/src/database/accessors.rs

```diff
@@ -201,7 +202,10 @@ pub async fn pull_and_update_deposit_with_retry(
         if update.is_unnecessary(&deposit_entry) {
             return Ok(deposit_entry);
         }
-        if !is_trusted_key && deposit_entry.status != Status::Pending {
+        let is_valid_untrusted_status_update = deposit_entry.status == Status::Pending
+            || update.event.status == StatusEntry::Accepted
+                && deposit_entry.status == Status::Accepted;
```

Comment:
I think this needs to be
```rust
let is_valid_untrusted_status_update = update.event.status == StatusEntry::Accepted &&
    (deposit_entry.status == Status::Pending || deposit_entry.status == Status::Accepted);
```

Well, maybe we need to allow an update if the current status is `Reprocessing` too, I forget if we actually use that state.

Reply:
Hmm, yeas, such condition looks better. About `Reprocessing` I think we don't really use it but not sure.

Reply:
[840d818](https://github.com/stacks-sbtc/sbtc/pull/1653/commits/840d8181e1a41903b044c4919c08ce3e60503272)

Reply:
Okay, I just a quick read through and it doesn't look like we use it. Would you mind checking too? We can remove it in a separate PR.

Reply:
I don't see any usages of this. Opened issue for it #1678